src/View/Helper/IdGeneratorTrait.php

```diff
@@ -59,7 +59,21 @@ protected function _clearIds()
     protected function _id($name, $val)
     {
         $name = $this->_domId($name);
+        $suffix = $this->_idSuffix($val);
 
+        return trim($name . '-' . $suffix, '-');
```

Comment:
Why do we need trim() here?

Reply:
It is here in case the fieldname starts or the value of the option ends with a dash. The HTML id attribute must not start with a dash, hence the trim. Might break some backwards compatibility when we remove it if someone uses field names starting with dashes.